keras_cv/layers/regularization/drop_path.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class DropPath(tf.keras.layers.Layer):
+    """
+    Implements the DropPath layer. Some samples from the batch are randomly
+    dropped during training with dropping probability `rate`. Note that
+    this layer drop individual samples within a batch and not the entire batch.
+
+    References:
+        - [FractalNet](https://arxiv.org/abs/1605.07648v4).
+        - [rwightman/pytorch-image-models](https://tinyurl.com/timm-droppath)
+
+    Args:
+        rate: float, the probability of the residual branch being dropped.
+
+    Usage:
+    `DropPath` can be used in any network as follows:
+    ```python
+
+    # (...)
+    input = tf.ones((1, 3, 3, 1), dtype=tf.float32)
+    residual = tf.keras.layers.Conv2D(1, 1)(input)
+    output = keras_cv.layers.DropPath()(input)
+    # (...)
+    ```
+    """
+
+    def __init__(self, rate=0.5, **kwargs):
+        super().__init__(**kwargs)
+        self.rate = rate
+
+    def call(self, x, training=None):
+        if self.rate == 0.0 or not training:
+            return x
+        else:
+            keep_prob = 1 - self.rate
+            drop_map_shape = (x.shape[0],) + (1,) * (len(x.shape) - 1)
+            drop_map = tf.keras.backend.random_bernoulli(drop_map_shape, p=keep_prob)
```

Comment:
can we seed this?

Reply:
If so, the `seed` should be set to what's passed to the constructor I guess.

Reply:
Nevermind, your https://github.com/keras-team/keras-cv/pull/480#discussion_r903113941 clears this.